model/src/test/kotlin/com/testerum/model/resources/http/response/XmlBodyHttpResponseTest.kt

```diff
@@ -0,0 +1,188 @@
+import com.testerum.model.resources.http.response.xml_body.XmlRootJsObject
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+
+internal class XmlBodyHttpResponseTest {
+
+    @Test
+    fun getAttributeValue() {
+        val xml = """
+            <bookstore>
+              <book>
+                <title lang="en">Harry Potter</title>
+                <price>29.99</price>
+              </book>
+
+              <book>
+                <title lang="nl">Learning XML</title>
+                <price>39.95</price>
+              </book>
+            </bookstore>
+        """.trimIndent()
+
+        val expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.bookstore.book[0].title.getAttr("lang")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("en", expressionResult)
+
+        val expressionResult1 = ExpressionEvaluator.evaluate(
+            """xmlBody.bookstore.book[1].title.getAttr("lang")""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("nl", expressionResult1)
+    }
+
+    @Test
+    fun getText() {
+        val xml = """
+            <bookstore>
+              <book>
+                <title lang="en">Harry Potter</title>
+                <price>29.99</price>
+              </book>
+
+              <book>
+                super tare
+                <title lang="nl">Learning XML</title>
+                clar ca da
+                <price>39.95</price>
+                merge si asa
+              </book>
+            </bookstore>
+        """.trimIndent()
+
+        var expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.bookstore.book[1].getText()""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("super tare clar ca da merge si asa", expressionResult)
+
+        expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.bookstore.book[1].getText(1)""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("clar ca da", expressionResult)
+
+        expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.bookstore.book[1].getText(2)""",
+            mapOf("xmlBody" to XmlRootJsObject(xml))
+        )
+        Assertions.assertEquals("merge si asa", expressionResult)
+    }
+
+    @Test
+    fun getChildMethodOnXmlNode() {
+        val xml = """
+            <bookstore>
+              <book>
+                <title lang="en">Harry Potter</title>
+                <price>29.99</price>
+              </book>
+
+              <book>
+                <title lang="nl">Learning XML</title>
+                <price>39.95</price>
+              </book>
+            </bookstore>
+        """.trimIndent()
+
+        val expressionResult = ExpressionEvaluator.evaluate(
+            """xmlBody.bookstore.getChild("book")[0].getChild("title").getAttr("lang")""",
```

Comment:
Let's use a child name (element name) that is one of the functions, for example ``xpath``. This will ensure that we can access such elements.